sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py

```diff
@@ -195,6 +195,90 @@ def query_batch(
             raise_with=LogsQueryError,
         )
 
+    @distributed_trace
+    def query_resource(
+        self,
+        resource_id: str,
+        query: str,
+        *,
+        timespan: Optional[Union[
+            timedelta, Tuple[datetime, timedelta], Tuple[datetime, datetime]]
+        ],
+        **kwargs: Any
+        ) -> Union[LogsQueryResult, LogsQueryPartialResult]:
+        """Execute a Kusto query on a resource.
+
+        Returns all the Azure Monitor logs matching the given Kusto query for an Azure resource.
+
+        :param resource_id: The identifier of the resource. The expected format is
+         '/subscriptions/<sid>/resourceGroups/<rg>/providers/<providerName>/<resourceType>/<resourceName>'.
+        :type resource_id: str
+        :param query: The Kusto query. Learn more about the `Kusto query syntax
+         <https://docs.microsoft.com/azure/data-explorer/kusto/query/>`_.
+        :type query: str
+        :keyword timespan: Required. The timespan for which to query the data. This can be a timedelta,
```

Comment:
docstring says required, but it's optional in method signature

Reply:
For required keyword-only arguments that can also be None, is `Optional` still the recommended way to define this?